map/Region.dhall

```diff
@@ -0,0 +1,257 @@
+                    , eu-north-1 = "eu-north-1"
+                    , eu-south-1 = "eu-south-1"
+                    , eu-west-1 = "eu-west-1"
+                    , eu-west-2 = "eu-west-2"
+                    , eu-west-3 = "eu-west-3"
+                    , me-south-1 = "me-south-1"
+                    , sa-east-1 = "sa-east-1"
+                    , us-gov-east-1 = "us-gov-east-1"
+                    , us-gov-west-1 = "us-gov-west-1"
+                    , us-east-1 = "us-east-1"
+                    , us-east-2 = "us-east-2"
+                    , us-west-1 = "us-west-1"
+                    , us-west-2 = "us-west-2"
+                    }
+                    region
+
+          in  render
+      , human =
+          let render
+              : Region → Text
+              = λ(region : Region) →
+                  merge
+                    { af-south-1 = "Africa (Cape Town)"
+                    , ap-east-1 = "Asia Pacific (Hong Kong)"
+                    , ap-northeast-1 = "Asia Pacific (Tokyo)"
+                    , ap-northeast-2 = "Asia Pacific (Sydney)"
+                    , ap-northeast-3 = "Asia Pacific (Osaka-Local)"
+                    , ap-south-1 = "Asia Pacific (Mumbai)"
+                    , ap-southeast-1 = "Asia Pacific (Singapore)"
+                    , ap-southeast-2 = "Asia Pacific (Sydney)"
+                    , ca-central-1 = "Canada (Central)"
+                    , cn-north-1 = "China (Beijing)"
+                    , cn-northwest-1 = "China (Ningxia)"
+                    , eu-central-1 = "Europe (Frankfurt)"
+                    , eu-north-1 = "Europe (Stockholm)"
+                    , eu-south-1 = "Europe (Milan)"
+                    , eu-west-1 = "Europe (Ireland)"
+                    , eu-west-2 = "Europe (London)"
+                    , eu-west-3 = "Europe (Paris)"
+                    , me-south-1 = "Middle East (Bahrain)"
+                    , sa-east-1 = "South America (São Paulo)"
+                    , us-gov-east-1 = "GovCloud (US-East)"
+                    , us-gov-west-1 = "GovCloud (US-West)"
+                    , us-east-1 = "US East (N. Virginia)"
+                    , us-east-2 = "US East (Ohio)"
+                    , us-west-1 = "US West (N. California)"
+                    , us-west-2 = "US West (Oregon)"
+                    }
+                    region
+
+          in  render
+      , availabilityZones =
+          let Params =
+                { Type =
+                    { newCustomer : { ap-northeast-1 : Bool, us-west-1 : Bool }
+                    }
+                , default = { ap-northeast-1 = True, us-west-1 = True }
+                }
+
+          let render
+              : Region → Params.Type → List Text
+              =
+                {- newCustomer flag is there because on aws new customers can
+                -- only access some of the availability zones.
+                -}
+                λ(region : Region) →
+                λ(_params : Params.Type) →
```

Comment:
prefer `\(_params : Params.Type) -> \(region : Region) ->` instead of the other way around so that the customer can provide only the `Params` and then call the result whenever they need to make a call on a region